samples/workload/tensorflow/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2019 Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+cmake_minimum_required (VERSION 3.14)
+
+project(tflite_wasm)
+
+include(ExternalProject)
+
+# Only Download
+ExternalProject_Add(tensorflow-download
+    PREFIX tensorflow
+    GIT_REPOSITORY https://github.com/tensorflow/tensorflow.git
+    GIT_TAG        v2.13.0-rc2
+    GIT_PROGRESS   ON
+    SOURCE_DIR     ${CMAKE_CURRENT_SOURCE_DIR}/tensorflow
+    UPDATE_COMMAND git restore .
+    PATCH_COMMAND  git apply ${CMAKE_CURRENT_SOURCE_DIR}/tflite_bazel_building.patch
+                     && ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_SOURCE_DIR}/bazelrc.user ${CMAKE_CURRENT_SOURCE_DIR}/tensorflow/.bazelrc.user
+                     && ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_SOURCE_DIR}/absl_em_asm_int.patch ${CMAKE_CURRENT_SOURCE_DIR}/tensorflow/third_party/absl
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND  ""
+    INSTALL_COMMAND ""
+    TEST_COMMAND   ""
+)
+
+ExternalProject_Add(mobilenet-download
+    URL  https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz
```

Comment:
How about changing this line to:
```cmake
    URL https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02
    DOWNLOAD_NAME mobilenet_v1_1.0_224_quant.tgz
```
The error below was reported in my building (and was fixed after using above lines):
```
CMake Error at /usr/share/cmake-3.22/Modules/ExternalProject.cmake:2806 (message):
  At least one entry of URL is a path (invalid in a list)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3716 (_ep_add_download_command)
  CMakeLists.txt:27 (ExternalProject_Add)
```

Reply:
it seems *https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02* is a directory not an valid item to download

```
$ wget https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02
--2023-11-07 05:59:12--  https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02
Resolving child-prc.intel.com (child-prc.intel.com)... 10.239.120.56
Connecting to child-prc.intel.com (child-prc.intel.com)|10.239.120.56|:913... connected.
Proxy request sent, awaiting response... 403 Forbidden
2023-11-07 05:59:13 ERROR 403: Forbidden.
```

And `DOWNLOAD_NAME` is 
> File name to use for the downloaded file. If not given, the end of the URL is used to determine the file name. ***This option is rarely needed,***

BTW, the original works fine with me

```
vscode ➜ .../wasm-micro-runtime/samples/workload/tensorflow (tflite_2.13) $ cmake --build build/ --target mobilenet-download
[ 12%] Creating directories for 'mobilenet-download'
[ 25%] Performing download step (download, verify and extract) for 'mobilenet-download'
-- Downloading...
   dst='/workspaces/wasm-micro-runtime/samples/workload/tensorflow/build/mobilenet-download-prefix/src/mobilenet_v1_1.0_224_quant.tgz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz'
-- Downloading... done
-- extracting...
     src='/workspaces/wasm-micro-runtime/samples/workload/tensorflow/build/mobilenet-download-prefix/src/mobilenet_v1_1.0_224_quant.tgz'
     dst='/workspaces/wasm-micro-runtime/samples/workload/tensorflow/mobilenet'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 37%] No update step for 'mobilenet-download'
[ 50%] No patch step for 'mobilenet-download'
[ 62%] No configure step for 'mobilenet-download'
[ 75%] No build step for 'mobilenet-download'
[ 87%] Performing install step for 'mobilenet-download'
[100%] Completed 'mobilenet-download'
[100%] Built target mobilenet-download
```

Reply:
URL + DOWNLOAD_NAME can be treated as the whole address, no need to make sure that `wget URL` works, it should be enough that `wget URL + / + DOWNLOAD_NAME` works. There is another sample in bwa workload:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/samples/workload/bwa/CMakeLists.txt#L60-L62

I am not sure why the building error is reported if using `URL  https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz`

Reply:
After trying, this patch works for me to download mobilenet (URL is unchanged, just add another line DOWNLOAD_NAME):
```C
diff --git a/samples/workload/tensorflow/CMakeLists.txt b/samples/workload/tensorflow/CMakeLists.txt
index 7d6ea166..df4be1ef 100644
--- a/samples/workload/tensorflow/CMakeLists.txt
+++ b/samples/workload/tensorflow/CMakeLists.txt
@@ -26,6 +26,7 @@ ExternalProject_Add(tensorflow-download

 ExternalProject_Add(mobilenet-download
     URL  https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz
+    DOWNLOAD_NAME mobilenet_v1_1.0_224_quant.tgz
     DOWNLOAD_EXTRACT_TIMESTAMP ON
     DOWNLOAD_NO_PROGRESS YES
     SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/mobilenet
```